spec/Classes.tex

```diff
@@ -587,41 +526,43 @@ \section{Class New}
-A class constructor is a method with the same name as the class.
-It is invoked by the \chpl{new} operator, where the
-class name and constructor arguments are preceded with the
-\chpl{new} keyword.
-
-When the constructor is called, memory is allocated to store
-a class instance, the instance undergoes default initialization, and
-then the constructor method is invoked on this newly-created
-instance.
-
-If the program declares a class constructor method,
-it is a user-defined constructor.
-If the program declares no constructors for a class,
-a compiler-generated constructor for that class is created automatically.
-
-\subsection{User-Defined Constructors}
-\label{User_Defined_Constructors}
-\index{classes!constructors!user-defined}
-\index{user-defined constructors}
-\index{constructors!user-defined}
-
-A user-defined constructor is a constructor method explicitly declared
-in the program.  A constructor declaration has the same
-syntax as a method declaration, except that the name of the function matches
-the name of the class, and there is no return type specifier.
-
-When a constructor is called, the usual function resolution mechanism
-(\rsec{Function_Resolution}) is applied to determine which
-user-defined constructor to invoke.
+\section{Class Initializers}
+\label{Class_Initializers}
+\index{classes!initializers}
+\index{initializers}
+
+A \chpl{new} expression allocates memory for the desired class and invokes
+an {\em initializer} method on the uninitialized memory, passing any arguments
+following the class name. An initializer is implemented by a method named
+\chpl{init} and is responsible for initializing the fields of the class.
+
+Any initializers declared in a program are \emph{user-defined} initializers. If
+the program declares no initializers for a class, the compiler must generate an
+initializer for that class based on the types and initialization expressions of
+fields defined by that class.
+
+\subsection{User-Defined Initializers}
+\label{User_Defined_Initializers}
+\index{classes!initializers!user-defined}
+\index{user-defined initializers}
+\index{initializers!user-defined}
+
+A user-defined initializer is an initializer method explicitly declared in the
+program.  An initializer declaration has the same syntax as a method
+declaration, with the restrictions that the name of the method must be
+\chpl{init} and there must not be a return type specifier. When an initializer
+is called, the usual function resolution mechanism (\rsec{Function_Resolution})
+is applied with the exception that an initializer may not be virtually
+dispatched.
+
+A user-defined initializer is responsible for initializing all fields. An
+initializer may omit initialization of fields, but all fields that are
+initialized must be initialized in declaration order.
+
+Initializers for generic classes (\rsec{Generic_Types}) handle \chpl{type} and
+\chpl{param} fields without defaults differently and may need to satisfy
```

Comment:
I think you should mention fields without types or default values here as well.  Otherwise a user might not think to check the generics section if they only had generic var/const fields